common-codegen/src/main/java/io/vrap/rmf/codegen/common/generator/extensions/MethodExtension.java

```diff
@@ -0,0 +1,35 @@
+package io.vrap.rmf.codegen.common.generator.extensions;
+
+import io.vrap.rmf.codegen.common.processor.annotations.ExtensionMethod;
+import io.vrap.rmf.codegen.common.processor.annotations.ModelExtension;
+import io.vrap.rmf.raml.model.resources.Method;
+import io.vrap.rmf.raml.model.responses.Body;
+import io.vrap.rmf.raml.model.types.AnyType;
+import io.vrap.rmf.raml.model.types.TypedElement;
+import io.vrap.rmf.raml.model.types.impl.TypesFactoryImpl;
+
+@ModelExtension(extend = Method.class)
+public class MethodExtension {
+
+
+    @ExtensionMethod
+    public AnyType getReturnType(final Method method) {
+        return method.getResponses().stream()
+                .filter(r -> r.getStatusCode().equals("200"))
+                .filter(response -> (response.getBodies() != null) && (!response.getBodies().isEmpty()))
+                .findFirst()
+                .map(method1 -> method1.getBodies().get(0))
+                .map(TypedElement::getType)
+                .orElse(TypesFactoryImpl.eINSTANCE.createNilType());
```

Comment:
I replaced your custom type with a proper RMF type here. Looks like I should give you a better introduction into RMF/EMF after your vacation 😄